src/dsm/headers/Graph.hpp

```diff
@@ -726,6 +732,20 @@ namespace dsm {
     return &(streetIt->second);
   }
 
+  template <typename Id, typename Size>
+    requires(std::unsigned_integral<Id> && std::unsigned_integral<Size>)
+  const std::unique_ptr<Street<Id, Size>>* Graph<Id, Size>::oppositeStreet(
+      Id streetId) const {
+    if (!m_streets.contains(streetId)) {
+      throw std::invalid_argument(
+          buildLog(std::format("Street with id {} does not exist: maybe it has changed "
+                               "id once called buildAdj.",
+                               streetId)));
+    }
+    const auto& nodePair = m_streets.at(streetId)->nodePair();
+    return this->street(nodePair.second, nodePair.first);
```

Comment:
The street is not granted to exist, right? In that case it would return a nullptr. Maybe we could use an optional?